test/tests/serial/mail_with_ldap.bats

```diff
@@ -1,245 +1,314 @@
+  _run_in_container_bash 'nc -w 1 0.0.0.0 143 < /tmp/docker-mailserver-test/auth/imap-ldap-auth.txt'
   assert_success
 }
 
-@test "checking postfix: ldap config overwrites success" {
-  run docker exec mail_with_ldap /bin/sh -c "grep 'server_host = ldap' /etc/postfix/ldap-users.cf"
-  assert_success
+@test "dovecot: ldap mail delivery works" {
+  _should_successfully_deliver_mail_to "some.user@${FQDN_LOCALHOST_A}" "/var/mail/${FQDN_LOCALHOST_A}/some.user/new/"
 
-  run docker exec mail_with_ldap /bin/sh -c "grep 'start_tls = no' /etc/postfix/ldap-users.cf"
-  assert_success
+  # Should support delivering to a local recipient with a different domain (and disjoint mail location):
+  # NOTE: Mail is delivered to location defined in `.ldif` (an account config setting, either `mailHomeDirectory` or `mailStorageDirectory`).
+  # `some.other.user` has been configured to use a mailbox domain different from it's address domain part, hence the difference here:
+  _should_successfully_deliver_mail_to "some.other.user@${FQDN_LOCALHOST_B}" "/var/mail/${FQDN_LOCALHOST_A}/some.other.user/new/"
+}
 
-  run docker exec mail_with_ldap /bin/sh -c "grep 'search_base = ou=people,dc=localhost,dc=localdomain' /etc/postfix/ldap-users.cf"
-  assert_success
+@test "dovecot: ldap config overwrites success" {
+  local LDAP_SETTINGS_DOVECOT=(
+    "uris = ldap://${FQDN_LDAP}"
+    'tls = no'
+    'base = ou=people,dc=localhost,dc=localdomain'
+    'dn = cn=admin,dc=localhost,dc=localdomain'
+  )
+
+  for LDAP_SETTING in "${LDAP_SETTINGS_DOVECOT[@]}"; do
+    _run_in_container grep "${LDAP_SETTING%=*}" /etc/dovecot/dovecot-ldap.conf.ext
+    assert_output "${LDAP_SETTING}"
+    assert_success
+  done
+}
 
-  run docker exec mail_with_ldap /bin/sh -c "grep 'bind_dn = cn=admin,dc=localhost,dc=localdomain' /etc/postfix/ldap-users.cf"
+# Requires ENV `POSTMASTER_ADDRESS`
+# NOTE: Not important to LDAP feature tests?
+@test "dovecot: postmaster address" {
+  _run_in_container grep "postmaster_address = postmaster@${FQDN_LOCALHOST_A}" /etc/dovecot/conf.d/15-lda.conf
   assert_success
+}
 
-  run docker exec mail_with_ldap /bin/sh -c "grep 'server_host = ldap' /etc/postfix/ldap-groups.cf"
-  assert_success
+# NOTE: `target/scripts/startup/setup.d/dovecot.sh` should prevent enabling the quotas feature when using LDAP:
+@test "dovecot: quota plugin is disabled" {
+  # Dovecot configs have not enabled the quota plugins:
+  _run_in_container grep "\$mail_plugins quota" /etc/dovecot/conf.d/10-mail.conf
+  assert_failure
+  _run_in_container grep "\$mail_plugins imap_quota" /etc/dovecot/conf.d/20-imap.conf
+  assert_failure
 
-  run docker exec mail_with_ldap /bin/sh -c "grep 'start_tls = no' /etc/postfix/ldap-groups.cf"
+  # Dovecot Quota config only present with disabled extension:
+  _run_in_container [ -f /etc/dovecot/conf.d/90-quota.conf ]
+  assert_failure
+  _run_in_container [ -f /etc/dovecot/conf.d/90-quota.conf.disab ]
```

Comment:
While absolutely fine, I in Bash we should prefer

```suggestion
  _run_in_container [[ -f /etc/dovecot/conf.d/90-quota.conf ]]
  assert_failure
  _run_in_container [[ -f /etc/dovecot/conf.d/90-quota.conf.disab ]]
```

This works because `[[` is a valid program, but _actually_, I'd try

```suggestion
  _run_in_container_bash '[[ -f /etc/dovecot/conf.d/90-quota.conf ]]'
  assert_failure
  _run_in_container_bash '[[ -f /etc/dovecot/conf.d/90-quota.conf.disab ]]'
```

to be absolutely sure nothing breaks.

Reply:
Sure, this was just referencing the manual SSL test which tested files in the container the same way.